middleware/ordered_group_test.go

```diff
@@ -190,3 +190,39 @@ func TestOrderedIDsGetOrder(t *testing.T) {
 		}
 	}
 }
+
+func TestOrderedIDsSlots(t *testing.T) {
+	o := newOrderedIDs()
+
+	noError(t, o.AddSlot("first", After))
+	noError(t, o.AddSlot("second", After))
+	noError(t, o.InsertSlot("fourth", "second", After))
+	noError(t, o.Insert(mockIder("third"), "second", After))
```

Comment:
Would be nice to have a swap positive and negative test case too.  Using test coverage with these are a great way to help more cases get hit.